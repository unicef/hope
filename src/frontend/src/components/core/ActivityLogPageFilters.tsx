import ViewModuleRoundedIcon from '@mui/icons-material/ViewModuleRounded';
import { Grid, MenuItem } from '@mui/material';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { useLocation, useNavigate } from 'react-router-dom';
import { AssigneeAutocomplete } from '@shared/autocompletes/AssigneeAutocomplete';
import { createHandleApplyFilterChange } from '@utils/utils';
import { FiltersSection } from './FiltersSection';
import { SearchTextField } from './SearchTextField';
import { SelectFilter } from './SelectFilter';

interface ActivityLogPageFiltersProps {
  filter;
  setFilter: (filter) => void;
  initialFilter;
  appliedFilter;
  setAppliedFilter: (filter) => void;
}
export function ActivityLogPageFilters({
  filter,
  setFilter,
  initialFilter,
  appliedFilter,
  setAppliedFilter,
}: ActivityLogPageFiltersProps): React.ReactElement {
  const { t } = useTranslation();
  const navigate = useNavigate();
  const location = useLocation();

  const { handleFilterChange, applyFilterChanges, clearFilter } =
    createHandleApplyFilterChange(
      initialFilter,
      navigate,
      location,
      filter,
      setFilter,
      appliedFilter,
      setAppliedFilter,
    );

  const handleApplyFilter = (): void => {
    applyFilterChanges();
  };

  const handleClearFilter = (): void => {
    clearFilter();
  };

  const modules = {
    program: 'Programme',
    household: 'Household',
    individual: 'Individual',
    grievanceticket: 'Grievance Tickets',
    paymentverificationplan: 'Payment Plan Payment Verification',
    targetpopulation: 'Target Population',
    registrationdataimport: 'Registration Data Import',
  };
  return (
    <FiltersSection
      clearHandler={handleClearFilter}
      applyHandler={handleApplyFilter}
    >
      <Grid container alignItems="center" spacing={3}>
        <Grid item xs={3}>
          <SearchTextField
            label={t('Search')}
            value={filter.search}
            onChange={(e) => handleFilterChange('search', e.target.value)}
            data-cy="filters-search"
          />
        </Grid>
        <Grid item xs={3}>
          <SelectFilter
            onChange={(e) => handleFilterChange('module', e.target.value)}
            label={t('Module')}
            value={filter.module}
            icon={<ViewModuleRoundedIcon />}
            SelectDisplayProps={{
              'data-cy': 'filters-residence-status',
            }}
            MenuProps={{
              'data-cy': 'filters-residence-status-options',
            }}
          >
            {Object.entries(modules)
              .sort()
              .map(([key, value]) => (
                <MenuItem key={key} value={key}>
                  {value}
                </MenuItem>
              ))}
          </SelectFilter>
        </Grid>
        <Grid item xs={3}>
          <AssigneeAutocomplete
            label={t('User')}
            filter={filter}
            name="userId"
            value={filter.userId}
            setFilter={setFilter}
            initialFilter={initialFilter}
            appliedFilter={appliedFilter}
            setAppliedFilter={setAppliedFilter}
          />
        </Grid>
      </Grid>
    </FiltersSection>
  );
}
