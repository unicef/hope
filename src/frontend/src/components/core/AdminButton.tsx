import ArrowCircleRightIcon from '@mui/icons-material/ArrowCircleRight';
import * as React from 'react';


interface AdminButtonProps {
    adminUrl: string;
    [key: string]: any;
}

export const AdminButton: React.FC<AdminButtonProps> = ({
    adminUrl,
    ...otherProps
}) => {
    if (adminUrl) {
        return <a href={adminUrl}><ArrowCircleRightIcon color="primary" {...otherProps}/></a>;
    }
    return null;
};
