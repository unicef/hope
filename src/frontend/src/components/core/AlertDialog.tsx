import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import * as React from 'react';
import { useEffect } from 'react';
import { useTranslation } from 'react-i18next';

export function AlertDialog({
  message,
  show,
}: {
  message: string;
  show: boolean;
}): React.ReactElement {
  const { t } = useTranslation();
  const [open, setOpen] = React.useState(false);
  const [oldShow, setOldShow] = React.useState(false);
  useEffect(() => {
    if (!oldShow && show) {
      setOpen(true);
    }
    setOldShow(show);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [show]);

  const handleClose = (): void => {
    setOpen(false);
  };

  return (
    <div>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">{t('Error')}</DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-description">
            {message}
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color="primary" autoFocus>
            OK
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
}
