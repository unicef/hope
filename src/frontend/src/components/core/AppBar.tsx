import { BusinessAreaSelect } from '@containers/BusinessAreaSelect';
import { GlobalProgramSelect } from '@containers/GlobalProgramSelect';
import { UserProfileMenu } from '@containers/UserProfileMenu';
import { useCachedMe } from '@hooks/useCachedMe';
import MenuIcon from '@mui/icons-material/Menu';
import TextsmsIcon from '@mui/icons-material/Textsms';
import { Box, Button } from '@mui/material';
import MuiAppBar from '@mui/material/AppBar';
import IconButton from '@mui/material/IconButton';
import Toolbar from '@mui/material/Toolbar';
import { styled } from '@mui/system';
import * as React from 'react';
import { MiśTheme, theme } from 'src/theme';

const StyledToolbar = styled(Toolbar)(() => ({
  display: 'flex',
  justifyContent: 'space-between',
}));

const StyledLink = styled('a')({
  textDecoration: 'none',
  color: '#e3e6e7',
});

interface AppBarProps {
  open: boolean;
}

interface StyledAppBarProps extends AppBarProps {
  theme: MiśTheme;
}
const StyledAppBar = styled(MuiAppBar)<StyledAppBarProps>(
  ({ theme: muiTheme, open }) => ({
    position: 'fixed',
    top: 0,
    zIndex: muiTheme.zIndex.drawer + 1,
    backgroundColor: muiTheme.palette.secondary.main,
    transition: muiTheme.transitions.create(['width', 'margin'], {
      easing: muiTheme.transitions.easing.sharp,
      duration: muiTheme.transitions.duration.leavingScreen,
    }),
    ...(open && {
      marginLeft: '270px',
      width: 'calc(100% - 270px)',
      transition: muiTheme.transitions.create(['width', 'margin'], {
        easing: muiTheme.transitions.easing.sharp,
        duration: muiTheme.transitions.duration.enteringScreen,
      }),
    }),
    ...(!open && {
      marginLeft: '0',
      width: '100%',
      transition: muiTheme.transitions.create(['width', 'margin'], {
        easing: muiTheme.transitions.easing.sharp,
        duration: muiTheme.transitions.duration.leavingScreen,
      }),
    }),
  }),
);

const StyledIconButton = styled(IconButton)<AppBarProps>(({ open }) => ({
  marginRight: 36,
  ...(open && {
    display: 'none',
  }),
}));

export const AppBar = ({ open, handleDrawerOpen }): React.ReactElement => {
  const { data: meData, loading: meLoading } = useCachedMe();
  const servicenow = `https://unicef.service-now.com/cc?id=sc_cat_item&sys_id=762ae3128747d91021cb670a0cbb35a7&HOPE - ${
    window.location.pathname.split('/')[2]
  }&Workspace: ${window.location.pathname.split('/')[1]} \n Url: ${
    window.location.href
  }`;

  if (meLoading) {
    return null;
  }
  return (
    <StyledAppBar theme={theme} open={open}>
      <StyledToolbar>
        <Box display="flex" alignItems="center" justifyContent="center">
          <Box ml={1}>
            <StyledIconButton
              edge="start"
              color="inherit"
              aria-label="open drawer"
              onClick={handleDrawerOpen}
              open={open}
            >
              <MenuIcon />
            </StyledIconButton>
          </Box>
          <Box display="flex" alignItems="center">
            <Box ml={6} data-cy="business-area-container">
              <BusinessAreaSelect />
            </Box>
            <Box ml={6} data-cy="global-program-filter-container">
              <GlobalProgramSelect />
            </Box>
          </Box>
        </Box>
        <Box display="flex" justifyContent="flex-end">
          <Button startIcon={<TextsmsIcon style={{ color: '#e3e6e7' }} />}>
            <StyledLink target="_blank" href={servicenow}>
              Support
            </StyledLink>
          </Button>
          <UserProfileMenu meData={meData} />
        </Box>
      </StyledToolbar>
    </StyledAppBar>
  );
};
