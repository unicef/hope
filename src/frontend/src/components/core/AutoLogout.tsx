import * as React from 'react';
import { useCallback, useEffect } from 'react';
import { useIdleTimer } from 'react-idle-timer';
import { AUTO_LOGOUT_MILLIS } from '../../config';

export function AutoLogout(): React.ReactElement {
  const handleOnIdle = (): void => {
    if (!localStorage.getItem('AUTHENTICATED')) {
      return;
    }
    window.location.assign('/api/logout');
    localStorage.removeItem('AUTHENTICATED');
  };

  //eslint-disable-next-line
  const { reset } = useIdleTimer({
    timeout: AUTO_LOGOUT_MILLIS,
    onIdle: handleOnIdle,
    debounce: 500,
  });

  const handleOnAction = useCallback((): void => {
    reset();
  }, [reset]);

  useEffect(() => {
    window.addEventListener('click', handleOnAction);
    window.addEventListener('mousemove', handleOnAction);

    return () => {
      window.removeEventListener('click', handleOnAction);
      window.removeEventListener('mousemove', handleOnAction);
    };
  }, [handleOnAction]);

  return <></>;
}
