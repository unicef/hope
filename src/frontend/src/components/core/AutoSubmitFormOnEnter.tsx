import { useFormikContext } from 'formik';
import * as React from 'react';
import { useEffect } from 'react';

export function AutoSubmitFormOnEnter(): React.ReactElement {
  const { submitForm } = useFormikContext();
  useEffect(() => {
    const handleEnter = (e: KeyboardEvent): void => {
      if (e.key === 'Enter') {
        submitForm();
      }
    };
    window.addEventListener('keypress', handleEnter);
    return () => {
      window.removeEventListener('keypress', handleEnter);
    };
  }, [submitForm]);
  return <></>;
}
