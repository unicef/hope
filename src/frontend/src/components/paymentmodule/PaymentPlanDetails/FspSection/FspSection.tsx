import { ContainerColumnWithBorder } from '@core/ContainerColumnWithBorder';
import { DividerLine } from '@core/DividerLine';
import { LabelizedField } from '@core/LabelizedField';
import { Box, Grid2 as Grid, Typography } from '@mui/material';
import { PaymentPlanDetail } from '@restgenerated/models/PaymentPlanDetail';
import { ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import { VolumeByDeliveryMechanismSection } from './VolumeByDeliveryMechanismSection';

interface FspSectionProps {
  baseUrl: string;
  paymentPlan: PaymentPlanDetail;
}

export const FspSection = ({ paymentPlan }: FspSectionProps): ReactElement => {
  const { t } = useTranslation();

  const { delivery_mechanisms, is_follow_up, financial_service_provider } =
    paymentPlan;
  const showFspDisplay = delivery_mechanisms?.length;
  const shouldDisableSetUpFsp = (): boolean => {
    if (paymentPlan.is_follow_up) {
      return false;
    }
    if (!paymentPlan.total_entitled_quantity_usd) {
      return true;
    }
    if (!isActiveProgram) {
      return true;
    }
    return false;
  };

  return (
    <Box m={5}>
      <ContainerColumnWithBorder>
        <Box
          display="flex"
          justifyContent="space-between"
          alignItems="center"
          mt={4}
        >
          <Typography variant="h6">{t('FSPs')}</Typography>
        </Box>
        <Grid container spacing={3}>
          <>
            <Grid
              key={`${deliveryMechanism?.name}-${financialServiceProvider?.name}`}
              size={{ xs: 3 }}
            >
              <LabelizedField
                label={deliveryMechanism?.name || '-'}
                value={financialServiceProvider?.name || '-'}
              />
            </Grid>
          </>
        </Grid>
        <DividerLine />
        <VolumeByDeliveryMechanismSection paymentPlan={paymentPlan} />
      </ContainerColumnWithBorder>
    </Box>
  );
};
