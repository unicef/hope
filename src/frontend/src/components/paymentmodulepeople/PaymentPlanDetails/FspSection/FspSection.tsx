import { Box, Button, Grid2 as Grid, Typography } from '@mui/material';
import { useTranslation } from 'react-i18next';
import { Link, useParams } from 'react-router-dom';
import { PaymentPlanQuery, PaymentPlanStatus } from '@generated/graphql';
import { ContainerColumnWithBorder } from '@core/ContainerColumnWithBorder';
import { DividerLine } from '@core/DividerLine';
import { LabelizedField } from '@core/LabelizedField';
import { useProgramContext } from '../../../../programContext';
import { VolumeByDeliveryMechanismSection } from './VolumeByDeliveryMechanismSection';
import { ReactElement } from 'react';

interface FspSectionProps {
  baseUrl: string;
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export function FspSection({
  baseUrl,
  paymentPlan,
}: FspSectionProps): ReactElement {
  const { t } = useTranslation();
  const { paymentPlanId } = useParams();
  const { isActiveProgram } = useProgramContext();

  const { deliveryMechanisms, isFollowUp } = paymentPlan;
  const showFspDisplay = deliveryMechanisms.length;
  const shouldDisableSetUpFsp = (): boolean => {
    if (paymentPlan.isFollowUp) {
      return false;
    }
    if (!paymentPlan.totalEntitledQuantityUsd) {
      return true;
    }
    if (!isActiveProgram) {
      return true;
    }
    return false;
  };

  return showFspDisplay ? (
    <Box m={5}>
      <ContainerColumnWithBorder>
        <Box
          display="flex"
          justifyContent="space-between"
          alignItems="center"
          mt={4}
        >
          <Typography variant="h6">{t('FSPs')}</Typography>
          {paymentPlan.status === PaymentPlanStatus.Locked && (
            <Button
              color="primary"
              variant="contained"
              component={Link}
              to={`/${baseUrl}/payment-module/${
                isFollowUp ? 'followup-payment-plans' : 'payment-plans'
              }/${paymentPlanId}/setup-fsp/edit`}
              disabled={!isActiveProgram}
            >
              {t('Edit FSP')}
            </Button>
          )}
        </Box>
        <Grid container spacing={3}>
          {deliveryMechanisms.map((el) => (
            <>
              <Grid key={el.name} size={{ xs: 3 }}>
                <LabelizedField label={el.name} value={el.fsp?.name} />
              </Grid>
              {el.chosenConfiguration && (
                <Grid key={el.chosenConfiguration} size={{ xs: 3 }}>
                  <LabelizedField
                    label="Configuration"
                    value={el.chosenConfiguration}
                  />
                </Grid>
              )}
            </>
          ))}
        </Grid>
        <DividerLine />
        <VolumeByDeliveryMechanismSection paymentPlan={paymentPlan} />
      </ContainerColumnWithBorder>
    </Box>
  ) : (
    <Box m={5}>
      <ContainerColumnWithBorder>
        <Box
          display="flex"
          justifyContent="space-between"
          alignItems="center"
          mt={4}
        >
          <Typography variant="h6">{t('FSPs')}</Typography>
          <Button
            color="primary"
            variant="contained"
            disabled={shouldDisableSetUpFsp()}
            data-cy="button-set-up-fsp"
            component={Link}
            to={`/${baseUrl}/payment-module/${
              isFollowUp ? 'followup-payment-plans' : 'payment-plans'
            }/${paymentPlanId}/setup-fsp/create`}
          >
            {t('Set up FSP')}
          </Button>
        </Box>
      </ContainerColumnWithBorder>
    </Box>
  );
}
