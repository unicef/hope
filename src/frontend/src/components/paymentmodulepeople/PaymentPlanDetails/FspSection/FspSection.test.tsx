import * as React from 'react';
import { fakeApolloPaymentPlan } from '../../../../../fixtures/paymentmodule/fakeApolloPaymentPlan';
import { render } from '../../../../testUtils/testUtils';
import { FspSection } from './FspSection';
import { fakeBaseUrl } from '../../../../../fixtures/core/fakeBaseUrl';

describe('components/paymentmodule/PaymentPlanDetails/FspSection', () => {
  it('should render Set Up FSP', () => {
    const { container } = render(
      <FspSection
        baseUrl={fakeBaseUrl}
        paymentPlan={{ ...fakeApolloPaymentPlan, deliveryMechanisms: [] }}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render Edit FSP', () => {
    const { container } = render(
      <FspSection baseUrl={fakeBaseUrl} paymentPlan={fakeApolloPaymentPlan} />,
    );
    expect(container).toMatchSnapshot();
  });
});
