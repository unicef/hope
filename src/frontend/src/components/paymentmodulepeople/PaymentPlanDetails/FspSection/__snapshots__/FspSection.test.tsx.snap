// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/PaymentPlanDetails/FspSection should render Edit FSP 1`] = `
<div>
  <div
    class="MuiBox-root css-zwiamn"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 sc-blHHSb kPa-dfK css-1ps6pg7-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-tdpf08"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1j53xw6-MuiTypography-root"
        >
          FSPs
        </h6>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-dl1jos-MuiButtonBase-root-MuiButton-root"
          href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/payment-module/payment-plans/undefined/setup-fsp/edit"
          tabindex="0"
        >
          Edit FSP
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-acctgf-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-fAUdSK goVbUg"
              color="textSecondary"
            >
              Cash
            </span>
            <div
              data-cy="label-Cash"
            >
              <span
                class="sc-dntaoT dIDHTF"
                color="textSecondary"
              >
                Miranda Ltd
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-fAUdSK goVbUg"
              color="textSecondary"
            >
              In Kind
            </span>
            <div
              data-cy="label-In Kind"
            >
              <span
                class="sc-dntaoT dIDHTF"
                color="textSecondary"
              >
                Frazier-Watson
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <div>
            <span
              class="sc-fAUdSK goVbUg"
              color="textSecondary"
            >
              Transfer
            </span>
            <div
              data-cy="label-Transfer"
            >
              <span
                class="sc-dntaoT dIDHTF"
                color="textSecondary"
              >
                Bray Group
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-gtLWhw VFxTS"
      >
        <div
          class="sc-egkSDF cYYMIc"
        />
      </div>
      <div
        class="MuiBox-root css-j7qwjs"
      >
        <div
          class="sc-ghWlax bzkNxN"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1j53xw6-MuiTypography-root"
          >
            Volume by Delivery Mechanism
          </h6>
           
        </div>
        <div
          class="sc-kLhKbu iAQbEA"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-1bzpx2o-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              <div
                class="sc-ivxoEo dPHKrZ"
                color="#D8E1EE"
              >
                <div>
                  <span
                    class="sc-fAUdSK goVbUg"
                    color="textSecondary"
                  >
                    Cash (Miranda Ltd)
                  </span>
                  <div
                    data-cy="label-Cash (Miranda Ltd)"
                  >
                    <span
                      class="sc-dntaoT dIDHTF"
                      color="textSecondary"
                    >
                      0 PLN (0 USD)
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              <div
                class="sc-ivxoEo bA-DRh"
                color=" #d8d8d8"
              >
                <div>
                  <span
                    class="sc-fAUdSK goVbUg"
                    color="textSecondary"
                  >
                    In Kind (Frazier-Watson)
                  </span>
                  <div
                    data-cy="label-In Kind (Frazier-Watson)"
                  >
                    <span
                      class="sc-dntaoT dIDHTF"
                      color="textSecondary"
                    >
                      0 PLN (0 USD)
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              <div
                class="sc-ivxoEo eiTvYj"
                color="#003C8F"
              >
                <div>
                  <span
                    class="sc-fAUdSK goVbUg"
                    color="textSecondary"
                  >
                    Transfer (Bray Group)
                  </span>
                  <div
                    data-cy="label-Transfer (Bray Group)"
                  >
                    <span
                      class="sc-dntaoT dIDHTF"
                      color="textSecondary"
                    >
                      0 PLN (0 USD)
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container css-1e2bu2o-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div
                class="sc-ixGGxD cHDLrf"
              >
                <canvas
                  height="0"
                  role="img"
                  style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
                  width="0"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/paymentmodule/PaymentPlanDetails/FspSection should render Set Up FSP 1`] = `
<div>
  <div
    class="MuiBox-root css-zwiamn"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 sc-blHHSb kPa-dfK css-1ps6pg7-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-tdpf08"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1j53xw6-MuiTypography-root"
        >
          FSPs
        </h6>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-dl1jos-MuiButtonBase-root-MuiButton-root"
          data-cy="button-set-up-fsp"
          href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/payment-module/payment-plans/undefined/setup-fsp/create"
          tabindex="0"
        >
          Set up FSP
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
